### PR TITLE
11 design implementation of identityapi

### PR DESCRIPTION
**Pull request by filippyziak**

### CODE REVIEW

File: src/Services/Identity.API/src/Identity.API.Application/Features/ChangeUserEmail/ChangeUserEmailCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+            if (await _identityRepository.AnyUserWithEmailAsync(request.Email))
+                throw new DuplicateExistsException($"User with {request.Email} already exists");
+
+            user.SetEmail(request.Email);
+
+            _mapper.Map(user, userPersistenceModel);
+
+            if (!await _identityRepository.UpdateUserAsync(userPersistenceModel))
+                throw new DatabaseOperationException($"Error while updating user #{user.UserId}");
```
Comment (TomeQ2k) [inline src/Services/Identity.API/src/Identity.API.Application/Features/ChangeUserEmail/ChangeUserEmailCommandHandler.cs:53]:
```suggestion
                throw new DatabaseOperationException($"Error while changing email for user #{user.UserId}");
```

File: src/Services/Identity.API/src/Identity.API.Application/Features/ChangeUserPassword/ChangeUserPasswordCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            var passwordHash = _passwordHashService.HashPassword(request.Password, passwordSalt);
+
+            var user = _mapper.Map<User>(userPersistenceModel);
+
+            user.SetPassword(passwordSalt, passwordHash);
+
+            _mapper.Map(user, userPersistenceModel);
+            
+            if (!await _identityRepository.UpdateUserAsync(userPersistenceModel))
+                throw new DatabaseOperationException($"Error while updating user #{user.UserId}");
```
Comment (TomeQ2k) [inline src/Services/Identity.API/src/Identity.API.Application/Features/ChangeUserPassword/ChangeUserPasswordCommandHandler.cs:51]:
```suggestion
                throw new DatabaseOperationException($"Error while updating password for user #{user.UserId}");
```

File: src/Services/Identity.API/src/Identity.API.Application/Features/SigninUser/SigninUserCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+            if (!_passwordHashService.VerifyPasswordHash(request.Password,
+                    userPersistenceModel.PasswordSalt,
+                    userPersistenceModel.PasswordHash))
+                throw new InvalidCredentialsException("Invalid email or password");
+
+            var user = _mapper.Map<User>(userPersistenceModel);
+            var jwtToken = _tokenService.GenerateAuthenticationToken(user);
+
+            _logger.Info($"jwt token has been generated for #{user.UserId}");
```
Comment (TomeQ2k) [inline src/Services/Identity.API/src/Identity.API.Application/Features/SigninUser/SigninUserCommandHandler.cs:47]:
```suggestion
            _logger.Info($"JWT token has been generated for #{user.UserId}");
```

File: src/Services/Identity.API/src/Identity.API.Domain/Configuration/ServiceConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using Library.Shared.AppConfigs;
+
+namespace Identity.API.Domain.Configuration
+{
+    public record ServiceConfiguration
+    {
+        public DatabaseConfig DatabaseConfig { get; set; }
+        public JwtConfig JwtConfig { get; set; }
```
Comment (TomeQ2k) [inline src/Services/Identity.API/src/Identity.API.Domain/Configuration/ServiceConfiguration.cs:8]:
```suggestion
        public DatabaseConfig DatabaseConfig { get; init; }
        public JwtConfig JwtConfig { get; init; }
```

File: src/Services/Venue.API/_Scripts/Database/init.sql [hunk 5]
```diff
@@ -1,44 +1,15 @@
...
+    "UserId" bigserial NOT NULL PRIMARY KEY,
+    "Email" varchar(320) NOT NULL UNIQUE,
+    "PasswordHash" bytea,
+    "PasswordSalt" bytea,
+    "IsDeleted" boolean DEFAULT false NOT NULL,
+    "CreatedOn" timestamp without time zone DEFAULT (now() at time zone 'utc') NOT NULL,
+    "ModifiedOn" timestamp without time zone
 );
 
-CREATE TABLE IF NOT EXISTS "Venue"."Locations"
```
Comment (TomeQ2k) [inline src/Services/Venue.API/_Scripts/Database/init.sql:16]: Why you override this file!?

**Review by TomeQ2k — CHANGES_REQUESTED** [review]
- Make requested changes
- REMOVE CHANGES FOR FILE VENUE.API INIT.SQL SCRIPT
- Remove unused folders
- Write healthcheck for database connection in API layer
- remove unit tests project if its not used

File: src/Services/Venue.API/_Scripts/Database/init.sql (hunk 5 above)
Comment (filippyziak) [reply src/Services/Venue.API/_Scripts/Database/init.sql:16]: XDDDD

File: src/Libraries/Library.Shared.Models/Identity/Events/UserEmailChangedEvent.cs [hunk 6]
```diff
@@ -4,6 +4,6 @@ namespace Library.Shared.Models.Identity.Events
 {
     public record UserEmailChangedEvent : Event
     {
-        public UserEmailChangedEvent() => EventType = EventType.USER_EMAIL_CHANGED;
+        public UserEmailChangedEvent() => EventType = EventType.USER_CREATED;
```
[the remark this answers is not in the document]
Comment (filippyziak) [reply src/Libraries/Library.Shared.Models/Identity/Events/UserEmailChangedEvent.cs:7]: XDDDD

File: src/Services/Identity.API/src/Identity.API.Application/Services/JwtAuthTokenService.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                Subject = new ClaimsIdentity(claims),
+                Expires = DateTime.UtcNow.AddMinutes(_jwtConfig.TokenExpirationInMinutes),
+                SigningCredentials = credentials
+            };
+
+            var jwtTokenHandler = new JwtSecurityTokenHandler();
+
+            var token = jwtTokenHandler.CreateToken(tokenDescriber);
+
+            return jwtTokenHandler.WriteToken(token);
```
[the remark this answers is not in the document]
Comment (filippyziak) [reply src/Services/Identity.API/src/Identity.API.Application/Services/JwtAuthTokenService.cs:51]: ?

File: src/Services/Identity.API/src/Identity.API/Controllers/IdentityController.cs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+        [HttpPost("signin")]
+        public async Task<IActionResult> SigninUser(SigninUserCommand command)
+        {
+            _logger.Info($"Sending command: {nameof(SigninUserCommand)}");
+
+            var response = await _mediator.Send(command);
+
+            return this.CreateResponse(response);
+        }
```
[the remark this answers is not in the document]
Comment (filippyziak) [reply src/Services/Identity.API/src/Identity.API/Controllers/IdentityController.cs:44]: why?

File: src/Services/Venue.API/_Scripts/Database/init.sql (hunk 5 above)
Comment (filippyziak) [reply src/Services/Venue.API/_Scripts/Database/init.sql]: i really don't know how that happened

File: src/Services/Identity.API/src/Identity.API.Application/Services/JwtAuthTokenService.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (TomeQ2k) [reply src/Services/Identity.API/src/Identity.API.Application/Services/JwtAuthTokenService.cs:51]: I remember sth that there was the async overload of this method

File: src/Services/Venue.API/_Scripts/Database/init.sql (hunk 5 above)
Comment (TomeQ2k) [reply src/Services/Venue.API/_Scripts/Database/init.sql]: me too

File: src/Services/Identity.API/src/Identity.API.Infrastructure/Database/Repositories/IdentityRepository.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        public IdentityRepository(IdentityDbContext identityDbContext)
+        {
+            _identityDbContext = identityDbContext;
+        }
+
+        public async Task<UserPersistenceModel> FindUserAsync(long userId)
+            => await _identityDbContext.Users.FirstOrDefaultAsync(u => u.UserId == userId);
+
+
+        public async Task<UserPersistenceModel> FindUserAsync(string email)
```
Comment (TomeQ2k) [inline src/Services/Identity.API/src/Identity.API.Infrastructure/Database/Repositories/IdentityRepository.cs:21]:
```suggestion
            => await _identityDbContext.Users.FirstOrDefaultAsync(u => u.UserId == userId);

        public async Task<UserPersistenceModel> FindUserAsync(string email)
```

File: src/Services/Venue.API/_Scripts/Database/init.sql [hunk 10]
```diff
@@ -1,4 +1,5 @@
-﻿CREATE SCHEMA IF NOT EXISTS "Venue";
+﻿
+CREATE SCHEMA IF NOT EXISTS "Venue";
```
Comment (TomeQ2k) [inline src/Services/Venue.API/_Scripts/Database/init.sql:2]:
```suggestion
CREATE SCHEMA IF NOT EXISTS "Venue";
```

File: src/Services/Identity.API/src/Identity.API.Application/Services/JwtAuthTokenService.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (filippyziak) [reply src/Services/Identity.API/src/Identity.API.Application/Services/JwtAuthTokenService.cs:51]: I couldn't Find it maybe not in this class

**Pull request merged**